File: clients/java/Thief.java
import com.sun.jna.Pointer;

///Represents a single thief on the map.
class Thief extends Mappable
{
  public Thief(Pointer p)
  {
    super(p);
  }
  boolean validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.thiefs.length; i++)
    {
      if(BaseAI.thiefs[i].ID == ID)
      {
        ptr = BaseAI.thiefs[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

    //commands

  ///Allows a thief to display messages on the screen
  boolean thiefTalk(String message)
  {
    validify();
    return (Client.INSTANCE.thiefThiefTalk(ptr, message) == 0) ? false : true;
  }
  ///Commands a thief to move to a new location.
  boolean move(int x, int y)
  {
    validify();
    return (Client.INSTANCE.thiefMove(ptr, x, y) == 0) ? false : true;
  }
  ///Commands a thief to use a special on a location.
  boolean useSpecial(int x, int y)
  {
    validify();
    return (Client.INSTANCE.thiefUseSpecial(ptr, x, y) == 0) ? false : true;
  }

    //getters

  ///Unique Identifier
  public int getId()
  {
    validify();
    return Client.INSTANCE.thiefGetId(ptr);
  }
  ///X position of the object
  public int getX()
  {
    validify();
    return Client.INSTANCE.thiefGetX(ptr);
  }
  ///Y position of the object
  public int getY()
  {
    validify();
    return Client.INSTANCE.thiefGetY(ptr);
  }
  ///The owner of this thief.
  public int getOwner()
  {
    validify();
    return Client.INSTANCE.thiefGetOwner(ptr);
  }
  ///The type of this thief. This type refers to list of thiefTypes.
  public int getThiefType()
  {
    validify();
    return Client.INSTANCE.thiefGetThiefType(ptr);
  }
  ///Whether the thief is alive or not.
  public int getAlive()
  {
    validify();
    return Client.INSTANCE.thiefGetAlive(ptr);
  }
  ///How many more times this thief can use its special ability.
  public int getSpecialsLeft()
  {
    validify();
    return Client.INSTANCE.thiefGetSpecialsLeft(ptr);
  }
  ///The maximum number of times this thief can use its special ability.
  public int getMaxSpecials()
  {
    validify();
    return Client.INSTANCE.thiefGetMaxSpecials(ptr);
  }
  ///The remaining number of times this thief can move.
  public int getMovementLeft()
  {
    validify();
    return Client.INSTANCE.thiefGetMovementLeft(ptr);
  }
  ///The maximum number of times this thief can move.
  public int getMaxMovement()
  {
    validify();
    return Client.INSTANCE.thiefGetMaxMovement(ptr);
  }
  ///How many turns this thief is frozen for.
  public int getFrozenTurnsLeft()
  {
    validify();
    return Client.INSTANCE.thiefGetFrozenTurnsLeft(ptr);
  }

}

File: clients/csharp/Thief.cs
using System;
using System.Runtime.InteropServices;

/// <summary>
/// Represents a single thief on the map.
/// </summary>
public class Thief: Mappable
{

  public Thief()
  {
  }

  public Thief(IntPtr p)
  {
    ptr = p;
    ID = Client.thiefGetId(ptr);
    iteration = BaseAI.iteration;
  }

  public override bool validify()
  {
    if(iteration == BaseAI.iteration) return true;
    for(int i = 0; i < BaseAI.thiefs.Length; i++)
    {
      if(BaseAI.thiefs[i].ID == ID)
      {
        ptr = BaseAI.thiefs[i].ptr;
        iteration = BaseAI.iteration;
        return true;
      }
    }
    throw new ExistentialError();
  }

  #region Commands
  /// <summary>
  /// Allows a thief to display messages on the screen
  /// </summary>
  public bool thiefTalk(string message)
  {
    validify();
    return (Client.thiefThiefTalk(ptr, message) == 0) ? false : true;
  }
  /// <summary>
  /// Commands a thief to move to a new location.
  /// </summary>
  public bool move(int x, int y)
  {
    validify();
    return (Client.thiefMove(ptr, x, y) == 0) ? false : true;
  }
  /// <summary>
  /// Commands a thief to act on a location.
  /// </summary>
  public bool act(int x, int y)
  {
    validify();
    return (Client.thiefAct(ptr, x, y) == 0) ? false : true;
  }
  #endregion

  #region Getters
  /// <summary>
  /// Unique Identifier
  /// </summary>
  public new int Id
  {
    get
    {
      validify();
      int value = Client.thiefGetId(ptr);
      return value;
    }
  }

  /// <summary>
  /// X position of the object
  /// </summary>
  public new int X
  {
    get
    {
      validify();
      int value = Client.thiefGetX(ptr);
      return value;
    }
  }

  /// <summary>
  /// Y position of the object
  /// </summary>
  public new int Y
  {
    get
    {
      validify();
      int value = Client.thiefGetY(ptr);
      return value;
    }
  }

  /// <summary>
  /// The owner of this thief.
  /// </summary>
  public int Owner
  {
    get
    {
      validify();
      int value = Client.thiefGetOwner(ptr);
      return value;
    }
  }

  /// <summary>
  /// The type of this thief. This type refers to list of thiefTypes.
  /// </summary>
  public int ThiefType
  {
    get
    {
      validify();
      int value = Client.thiefGetThiefType(ptr);
      return value;
    }
  }

  /// <summary>
  /// Whether the thief is alive or not.
  /// </summary>
  public int Alive
  {
    get
    {
      validify();
      int value = Client.thiefGetAlive(ptr);
      return value;
    }
  }

  /// <summary>
  /// How many more times this thief can use its special ability.
  /// </summary>
  public int SpecialsLeft
  {
    get
    {
      validify();
      int value = Client.thiefGetSpecialsLeft(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum number of times this thief can use its special ability.
  /// </summary>
  public int MaxSpecials
  {
    get
    {
      validify();
      int value = Client.thiefGetMaxSpecials(ptr);
      return value;
    }
  }

  /// <summary>
  /// The remaining number of times this thief can move.
  /// </summary>
  public int MovementLeft
  {
    get
    {
      validify();
      int value = Client.thiefGetMovementLeft(ptr);
      return value;
    }
  }

  /// <summary>
  /// The maximum number of times this thief can move.
  /// </summary>
  public int MaxMovement
  {
    get
    {
      validify();
      int value = Client.thiefGetMaxMovement(ptr);
      return value;
    }
  }

  /// <summary>
  /// How many turns this thief is frozen for.
  /// </summary>
  public int FrozenTurnsLeft
  {
    get
    {
      validify();
      int value = Client.thiefGetFrozenTurnsLeft(ptr);
      return value;
    }
  }

  #endregion

  #region Properties
  #endregion
}
